src/v/kafka/server/handlers/produce.cc

```diff
@@ -462,6 +468,19 @@ produce_handler::handle(request_context ctx, ss::smp_service_group ssg) {
     produce_request request;
     request.decode(ctx.reader(), ctx.header().version);
 
+    if (ctx.metadata_cache().should_reject_writes()) {
+        thread_local static ss::logger::rate_limit rate(despam_interval);
```

Comment:
You pass in a context `rate_limit` that you manage, and that is what ties the different invocations of the log statement together. Before logging, it checks the `rate_limit` (timestamp + helper functions) to see if enough time since last log has elapsed. If _not,_ it increments a dropped counter and returns. If enough time _has_ passed (or it is first msg), then it logs normally. Basically, what you'd expect knowing these seastar devs love performance like we do.  😊